File: 2021/2021W27/PreppindataW27.py
import pandas as pd
import random

xlsx = pd.ExcelFile("F:/Data/PD 2021 Wk 27 Input.xlsx")

team = pd.read_excel(xlsx,sheet_name='Teams')
seeding = pd.read_excel(xlsx,sheet_name='Seeding')

teams = team['Seed'].tolist()
temp = []

for rnd in range(1,5):
    pick = random.choices(teams, weights=tuple(list(seeding[rnd])), k=1)[0]
    df = team[team['Seed']==pick].copy()
    df['Actual Pick'] = rnd
    temp.append(df)
    
    teams.remove(pick)
    seeding = seeding[seeding['Seed']!=pick].copy()
    
picked = pd.concat(temp)

remain = team[team['Seed'].isin(teams)].copy()
remain['Actual Pick'] = remain['Seed'].rank(ascending=True).astype(int)+rnd
    
final = pd.concat([picked, remain])[['Actual Pick', 'Seed', 'Team']].rename(columns={'Seed':'Original'})

File: 2021/2021W27/preppindataW27.R
library(magrittr)
library(readxl)

xlsx <- "F:/Data/PD 2021 Wk 27 Input.xlsx"

seeding <- read_excel(xlsx, sheet = 'Seeding') %>%
  `colnames<-`(c('Seed', paste0('Round',c(1:14)))) %>%
  merge(., read_excel(xlsx, sheet = 'Teams'), by='Seed') 

picked <- data.frame(`Actual Pick`=numeric(), Seed=numeric(), Team=character())

for(i in 1:4) {
  df <- sample_n(seeding %>% select(c('Seed', 'Team')), size=1, weight=seeding[[paste0('Round',i)]]) %>%
    mutate('Actual Pick' = i)
  seeding %<>% filter(`Seed` != df[['Seed']])
  picked <- rbind(picked, df)
}

final <- rbind(picked, seeding %>% 
                 select(c('Seed', 'Team')) %>%
                 mutate('Actual Pick' = rank(`Seed`) +i)) %>%
  rename('Original' = 'Seed') %>%
  select(c('Actual Pick', 'Original', 'Team'))

View(final)
